### ISSUE TITLE
Keras confusion matrix is not as per accuracy given by fitting following CNN model

### ISSUE THREAD

**Issue by vishavdhanjal** [keras-team/keras#11033]
PYTHON CODE FOR CNN:
model = Sequential()
model.add(Conv1D(16,kernel_size=3, input_shape=(300,1)))
model.add(Conv1D(32,kernel_size=3))
model.add(Flatten())
model.add(Dense(2000, activation='relu'))#, input_dim=300))
model.add(Dense(1000, activation='relu'))
model.add(Dense(500, activation='relu'))
model.add(Dense(250, activation='relu'))
model.add(Dense(5, activation='sigmoid'))
model.summary()
model.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics = ['accuracy'])

model.fit(trainX, trainY, batch_size = 10, epochs = 10, verbose=1,
validation_data=(testX, testY))

CONFUSION MATRIX IS NOT ACCURATE FOR CNN THOUGH FOR ANN IT IS ACCURATE IN TERMS OF ACCURACY.

**Comment by brge17** [comment]
Looks like you have made several mistakes.

``model.add(Dense(num_classes, activation="softmax"))``

``model.compile(... loss="categorical_crossentropy" ...)``

Should fix them.
